### ISSUE TITLE
Extension fails to load on gnome-shell 3.32 (Ubuntu 19.04)

### ISSUE THREAD

**Issue by jdknox** [ethanwharris/gnome-nvidia-extension#168]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

# Current Behavior

Can't get the extension to load for Gnome Shell 3.32 after updating to Ubuntu 19.04

# Failure Information (for bugs)

    gnome-shell[1407]: Extension "nvidiautil@ethanwharris" had error: TypeError: this._construct is not a function

## Context

* Nvidia-Settings Version (if used):
* Nvidia-smi version (if used):
* Operating System: Ubuntu 19.04

## Failure Logs

```
-- Logs begin at Mon 2019-04-15 13:28:53 EEST. --
Apr 23 18:22:39 - gnome-shell[1407]: Received error from DBus search provider org.gnome.seahorse.Application.desktop during GetResultMetas: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.gnome.Shell.SearchProvider2” on object at path /org/gnome/seahorse/Application
Apr 23 18:22:39 - gnome-shell[1407]: Wrong number of result metas returned by search provider org.gnome.seahorse.Application.desktop: expected 1 but got 0
Apr 23 18:22:42 - gnome-shell[1407]: Received error from DBus search provider org.gnome.seahorse.Application.desktop during GetResultMetas: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name :1.392 was not provided by any .service files
Apr 23 18:22:42 - gnome-shell[1407]: Wrong number of result metas returned by search provider org.gnome.seahorse.Application.desktop: expected 1 but got 0
Apr 23 18:22:43 - gnome-shell[1407]: Received error from DBus search provider org.gnome.seahorse.Application.desktop during GetResultMetas: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name :1.393 was not provided by any .service files
Apr 23 18:22:43 - gnome-shell[1407]: Wrong number of result metas returned by search provider org.gnome.seahorse.Application.desktop: expected 1 but got 0
Apr 23 18:22:44 - gnome-shell[1407]: Received error from DBus search provider org.gnome.seahorse.Application.desktop during GetResultMetas: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.gnome.Shell.SearchProvider2” on object at path /org/gnome/seahorse/Application
Apr 23 18:22:44 - gnome-shell[1407]: Wrong number of result metas returned by search provider org.gnome.seahorse.Application.desktop: expected 1 but got 0
Apr 23 18:22:44 - gnome-shell[1407]: g_environ_setenv: assertion 'value != NULL' failed
Apr 23 18:23:04 - org.gnome.Shell.desktop[1407]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x260000d
Apr 23 18:28:22 - org.gnome.Shell.desktop[1407]: Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
Apr 23 18:28:23 - org.gnome.Shell.desktop[1407]: current session already has an ibus-daemon.
Apr 23 18:28:24 - gnome-shell[1407]: g_dir_open_with_errno: assertion 'path != NULL' failed
Apr 23 18:28:24 - gnome-shell[1407]: g_filename_to_utf8: assertion 'opsysstring != NULL' failed
Apr 23 18:28:24 - gnome-shell[1407]: g_dir_open_with_errno: assertion 'path != NULL' failed
Apr 23 18:28:24 - gnome-shell[1407]: g_filename_to_utf8: assertion 'opsysstring != NULL' failed
Apr 23 18:28:24 - gnome-shell[1407]: Telepathy is not available, chat integration will be disabled.
Apr 23 18:28:24 - gnome-shell[1407]: Some code accessed the property 'translate' on the module 'translation'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Apr 23 18:28:24 - gnome-shell[1407]: Some code accessed the property 'init' on the module 'translation'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Apr 23 18:28:24 - gnome-shell[1407]: Some code accessed the property 'init' on the module 'icons'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Apr 23 18:28:24 - gnome-shell[1407]: Some code accessed the property 'Widget' on the module 'menu'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
Apr 23 18:28:24 - gnome-shell[1407]: Extension "prime-indicator@gnome-shell-exstensions.fffilo.github.com" had error: TypeError: this._construct is not a function
Apr 23 18:28:24 - gnome-shell[1407]: Extension "primelaunch-gnome@tomasm.tk" had error: SyntaxError: missing = in const declaration
Apr 23 18:28:24 - gnome-shell[1407]: Extension "nvidiautil@ethanwharris" had error: TypeError: this._construct is not a function
Apr 23 18:28:24 - gnome-shell[1407]: JS WARNING: [/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/appIcons.js 1054]: unreachable code after return statement
Apr 23 18:28:24 - gnome-shell[1407]: Extension "ubuntu-dock@ubuntu.com" had error: Error: Type name Gjs_DashToDock_MyDashActor is already registered
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-DEBUG] Registering StatusNotifierItem :1.101/StatusNotifierItem
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-DEBUG] Registering StatusNotifierItem :1.84/org/ayatana/NotificationItem/dropbox_client_1762
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-DEBUG] Registering StatusNotifierItem :1.97/org/ayatana/NotificationItem/software_update_available
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-DEBUG] Registering StatusNotifierItem :1.94/org/ayatana/NotificationItem/chrome_app_indicator_1
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-DEBUG] Registering StatusNotifierItem :1.94/org/ayatana/NotificationItem/chrome_app_indicator_2
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-DEBUG] Registering StatusNotifierItem :1.70/org/ayatana/NotificationItem/blueman
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:24 - gnome-shell[1407]: [AppIndicatorSupport-FATAL] unable to update overlay icon
Apr 23 18:28:25 - gnome-shell[1407]: Error looking up permission: GDBus.Error:org.freedesktop.portal.Error.NotFound: No entry for geolocation
Apr 23 18:28:26 - org.gnome.Shell.desktop[1407]: Window manager warning: Overwriting existing binding of keysym 35 with keysym 35 (keycode e).
Apr 23 18:28:26 - org.gnome.Shell.desktop[1407]: Window manager warning: Overwriting existing binding of keysym 32 with keysym 32 (keycode b).
Apr 23 18:28:26 - org.gnome.Shell.desktop[1407]: Window manager warning: Overwriting existing binding of keysym 39 with keysym 39 (keycode 12).
Apr 23 18:28:26 - org.gnome.Shell.desktop[1407]: Window manager warning: Overwriting existing binding of keysym 33 with keysym 33 (keycode c).
Apr 23 18:28:26 - org.gnome.Shell.desktop[1407]: Window manager warning: Overwriting existing binding of keysym 36 with keysym 36 (keycode f).
Apr 23 18:28:26 - org.gnome.Shell.desktop[1407]: Window manager warning: Overwriting existing binding of keysym 38 with keysym 38 (keycode 11).
Apr 23 18:28:26 - org.gnome.Shell.desktop[1407]: Window manager warning: Overwriting existing binding of keysym 37 with keysym 37 (keycode 10).
Apr 23 18:28:26 - org.gnome.Shell.desktop[1407]: Window manager warning: Overwriting existing binding of keysym 31 with keysym 31 (keycode a).
Apr 23 18:28:26 - org.gnome.Shell.desktop[1407]: Window manager warning: Overwriting existing binding of keysym 34 with keysym 34 (keycode d).
Apr 23 18:28:26 - gnome-shell[1407]: GNOME Shell started at Tue Apr 23 2019 18:28:24 GMT+0300 (EEST)
```

**Comment by jdknox** [comment]
This is related: https://github.com/micheleg/dash-to-dock/pull/881

**Comment by ethanwharris** [comment]
Thanks for the issue! Unfortunately we don't have a machine running 19.04 to test this on but please feel free to raise a PR if you are able to fix it :)

**Comment by jdknox** [comment]
I was about to submit a pull request with the changes I've been working on, but it might not work if you don't have access to Gnome Shell 3.32.  I think my changes work as far as Gnome is concerned, but for some reason my update breaks the GPU reporting functionality.  The temperature is not showing up in the extension, but is instead replaced by the GPU utilization percentage.  And the fan speed shows a constant 2002%.

I was hoping you might be able to spot where I messed something up...

![Screenshot from 2019-04-29 16-43-36](https://user-images.githubusercontent.com/926182/56900261-1028d800-6a9e-11e9-94ba-e5618b86fe0a.png)

**Comment by ethanwharris** [comment]
Hmm, everything looks good to me. The only thing that might be causing the problem is line 143 in extension.js which I think should say `super.setActive(active);` rather than `super._init(active);`. It's not a problem if we update the extension to just work for 3.32 since the old stuff will still be there. I'll see if I can get something with 3.32 on to test it in case the above doesn't work.

**Comment by jdknox** [comment]
I also found another issue in settingsProperties.js line 30 that should have been `return super.parse(lines);` instead of `return lines;`.

After those changes I can get the `nvidia-smi` processor to work, but the `nvidia-settings` processor is still having trouble parsing.

**Comment by jdknox** [comment]
Apparently I should have contacted the IT department at Reynholm Industries; once I updated my GPU driver and rebooted, everything works fine.

**Comment by ethanwharris** [comment]
Closed by #169
